Model/Sync/Catalog/Data.php

```diff
@@ -369,65 +378,69 @@ protected function prepareOptions($syncItems, $configIds, $productObjects)
      */
     public function attributeMapping(Product $item)
     {
-        $itemArray = [];
-        $mapAttributes = $this->mappingAttributes;
+        $itemAttributesData = [];
 
-        foreach ($mapAttributes as $key => $attr) {
+        foreach ($this->attributesMapping as $attributeKey => $attributeDetails) {
             try {
-                if ($key === 'gtins') {
-                    $value = $this->prepareGtinsData($attr, $item);
-                } elseif ($key === 'custom_properties') {
-                    $value = $this->prepareCustomProperties($attr, $item);
-                } elseif ($key === 'is_discontinued') {
-                    $value = false;
-                } else {
-                    if ($attr['default']) {
-                        $data = $item->getData($attr['attr_code']);
-
-                        if (isset($attr['type']) && $attr['type'] === 'url') {
-                            $data = $item->getProductUrl();
+                switch ($attributeKey) {
+                    case 'gtins':
+                        $value = $this->prepareGtinsData($attributeDetails, $item);
+                        break;
+                    case 'custom_properties':
+                        $value = $this->prepareCustomProperties($attributeDetails, $item);
+                        break;
+                    case 'is_discontinued':
+                        $value = false;
+                        break;
+                    case 'url':
+                        $value = $item->getProductUrl();
+                        break;
+                    case 'image_url':
+                        $value = $this->getProductImageUrl($item);
+                        break;
+                    default:
+                        if (!$attributeDetails['default'] && isset($attributeDetails['method']) && $attributeDetails['method']) {
+
+                            $configKey = isset($attributeDetails['attr_code']) && $attributeDetails['attr_code'] ?
+                                $attributeDetails['attr_code'] : '';
+
+                            $method = $attributeDetails['method'];
+                            $itemValue = $this->$method($item, $configKey);
+                            if ($itemValue) {
+                                $value = $itemValue;
+                            } elseif ($method == 'getProductPrice') {
+                                $value = 0.00;
+                            } else {
+                                $value = $itemValue;
+                            }
```

Comment:
I think we changed the logic now no? Isn't it should be:
```suggestion
                            if ($method == 'getProductPrice') {
                                $value = 0.00;
                            }  else {
                                $value = $itemValue;
                            }
```

Reply:
I think that your suggestion is the one changing the logic.
The one I wrote broke down the one-liner we had there.
According to what you wrote, all `getProductPrice` will get a value of `0.00`.

Reply:
You are right, my bad. maybe lets change it then to:
```
if (!$itemValue && $method == 'getProductPrice') {
    $value = 0.00;
}  else {
    $value = $itemValue;
}
```

the three ifs condition is confusing. wdyt?